src/Build/Instance/TaskRegistry.cs

```diff
@@ -1361,7 +1358,9 @@ private bool GetTaskFactory(TargetLoggingContext targetLoggingContext, ElementLo
                                 }
                                 finally
                                 {
+#if FEATURE_APPDOMAIN
```

Comment:
Not sure if it actually matters, but should this be outside the `finally` so the compiler/runtime doesn't have to build up infrastructure for finally?